projects/packages/jetpack-mu-wpcom/src/features/wpcom-themes/wpcom-themes.php

```diff
@@ -7,85 +7,363 @@
+			$customize_url = $theme->block_theme ? admin_url( 'site-editor.php' ) : wp_customize_url( $theme->slug );
+
+			$theme->customize_url = add_query_arg(
+				array(
+					'return' => urlencode( network_admin_url( 'theme-install.php', 'relative' ) ),
+				),
+				$customize_url
+			);
+		}
+
+		$theme->name        = wp_kses( $theme->name, $themes_allowedtags );
+		$theme->author      = wp_kses( $theme->author['display_name'], $themes_allowedtags );
+		$theme->version     = wp_kses( $theme->version, $themes_allowedtags );
+		$theme->description = wp_kses( $theme->description, $themes_allowedtags );
+
+		$theme->stars = wp_star_rating(
+			array(
+				'rating' => $theme->rating,
+				'type'   => 'percent',
+				'number' => $theme->num_ratings,
+				'echo'   => false,
+			)
+		);
+
+		$theme->num_ratings    = number_format_i18n( $theme->num_ratings );
+		$theme->preview_url    = set_url_scheme( $theme->preview_url );
+		$theme->compatible_wp  = is_wp_version_compatible( $theme->requires );
+		$theme->compatible_php = is_php_version_compatible( $theme->requires_php );
+
+		// @patched Add filter to update theme object.
+		apply_filters( 'wpcom_themes_api_theme_object', $theme );
+	}
+
+	wp_send_json_success( $api );
 }
-add_action( 'load-theme-install.php', 'wpcom_themes_show_banner' );
+add_action( 'wp_ajax_query-themes', 'wpcom_themes_ajax_query_themes', 1 );
 
 /**
- * Registers an "Appearance > Theme Showcase" menu.
+ * Remove the core themes save-wporg-username ajax action and replace it with our own,
+ * which does not check `current_user_can( 'install_themes' )`.
  */
-function wpcom_themes_add_theme_showcase_menu() {
-	if ( get_option( 'wpcom_admin_interface' ) !== 'wp-admin' ) {
-		return;
+function wpcom_themes_ajax_save_wporg_username() {
+	// Remove the core function
+	remove_action( 'wp_ajax_query-themes', 'wp_ajax_save_wporg_username' );
+
+	// Copy over the `wp_ajax_save_wporg_username` function.
+
+	// @patched Remove the capability check.
+
+	check_ajax_referer( 'save_wporg_username_' . get_current_user_id() );
+
+	$username = isset( $_REQUEST['username'] ) ? wp_unslash( $_REQUEST['username'] ) : false;
+
+	if ( ! $username ) {
+		wp_send_json_error();
 	}
 
-	$site_slug = wp_parse_url( home_url(), PHP_URL_HOST );
-	add_submenu_page(
-		'themes.php',
-		esc_attr__( 'Theme Showcase', 'jetpack-mu-wpcom' ),
-		__( 'Theme Showcase', 'jetpack-mu-wpcom' ),
-		current_user_can( 'switch_themes' ) ? 'switch_themes' : 'edit_theme_options',
-		"https://wordpress.com/themes/$site_slug?ref=wpcom-themes-menu"
+	wp_send_json_success( update_user_meta( get_current_user_id(), 'wporg_favorites', $username ) );
+}
+add_action( 'wp_ajax_save-wporg-username', 'wpcom_themes_ajax_save_wporg_username', 1 );
+
+/**
+ * Filter the theme object.
+ *
+ * @param object $theme The theme object.
+ * @return object The theme object.
+ */
+function wpcom_themes_api_theme_object( $theme ) {
```

Comment:
Should this get added to the endpoint itself?

Reply:
I left it as a filter (in `wpcom_themes_ajax_query_themes`) because that function is copied from core, and our code here needs to run after the body of that function runs.

The alternative would be updating the endpoint, but then we would need to alter the `wpcom_themes_ajax_query_themes` more, which could be harder to merge later if there are core changes. I added a note about that to [my how to merge comment](https://github.com/Automattic/jetpack/pull/41374#issuecomment-2651696266).